req.go

```diff
@@ -81,6 +91,139 @@ func (req *reqSocket) SetOption(name string, value interface{}) error {
 	return req.sck.SetOption(name, value)
 }
 
+type reqWriter struct {
+	ctx      context.Context
+	mu       sync.Mutex
+	conns    []*Conn
+	nextConn int
+	state    *reqState
+}
+
+type reqReader struct {
+	ctx   context.Context
```

Comment:
do we need this field?
it doesn't seem so.
(we can keep the signature of `newReqReader` as is, though)